### PR TITLE
ISPN-8725 Document REST server changes

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.jboss.org/browse/ISPN-8725

### CODE REVIEW

File: documentation/src/main/asciidoc/user_guide/server_protocols.adoc [hunk 1]
```diff
@@ -76,7 +76,7 @@ Choosing the right protocol depends on a number of factors.
 | Bulk ops           | [green]*Y* | [red]*N*        | [red]*N*
 | Transactions       | [red]*N*   | [red]*N*        | [red]*N*
 | Listeners          | [green]*Y* | [red]*N*        | [red]*N*
-| Query              | [green]*Y* | [red]*N*        | [red]*N*
+| Query              | [green]*Y* | [green]*Y*      | [red]*N*
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/user_guide/server_protocols.adoc]: This is lovely !

File: documentation/src/main/asciidoc/user_guide/server_protocols_rest.adoc [hunk 2]
```diff
@@ -12,18 +41,129 @@ Note: TLS/ALPN with JDK8 requires additional steps from the client perspective.
...
+* _application/x-java-object_
+* _application/octet-stream_
+* _application/x-www-form-urlencoded_
+* _text/plain_
+
+The following formats can be converted to/from the formats above:
+
+* __application/xml__
+* _application/json_
+* _application/x-jboss-marshalling_
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/user_guide/server_protocols_rest.adoc]: Do we want to add formats for Kryo and ProtoStuff marshalling?

**Comment by tristantarrant** [comment]
Merged, thanks

File: documentation/src/main/asciidoc/user_guide/server_protocols_rest.adoc (hunk 2 above)
Comment (gustavonalle) [reply documentation/src/main/asciidoc/user_guide/server_protocols_rest.adoc]: I think they should be handled client side only for now, as those two are not part of the server

**Pull request merged**